agent-nodejs/src/opamp/client-http.ts

```diff
@@ -0,0 +1,208 @@
+import { PartialMessage } from "@bufbuild/protobuf";
+import {
+  AgentDescription,
+  AgentToServer,
+  ServerToAgent,
+  ServerToAgentFlags,
+} from "./generated/opamp_pb";
+import { OpAMPClientHttpConfig, ResourceAttributeFromServer } from "./types";
+import {
+  keyValuePairsToOtelAttributes,
+  otelAttributesToKeyValuePairs,
+} from "./utils";
+import { uuidv7 } from "uuidv7";
+import axios, { AxiosInstance } from "axios";
+import { DetectorSync, IResource, Resource } from "@opentelemetry/resources";
+import { Attributes, context, diag } from "@opentelemetry/api";
+import {
+  SEMRESATTRS_SERVICE_INSTANCE_ID,
+  SEMRESATTRS_SERVICE_NAME,
+} from "@opentelemetry/semantic-conventions";
+import { suppressTracing } from "@opentelemetry/core";
+
+export class OpAMPClientHttp implements DetectorSync {
+  private config: OpAMPClientHttpConfig;
+  private OpAMPInstanceUidString: string;
+  private OpAMPInstanceUid: Uint8Array;
+  private nextSequenceNum: bigint = BigInt(0);
+  private httpClient: AxiosInstance;
+  private logger = diag.createComponentLogger({
+    namespace: "@odigos/opentelemetry-node/opamp",
+  });
+
+  // promise that we can resolve async later on, which the detect function can return
+  private resourcePromiseResolver?: (resourceAttributes: Attributes) => void;
+
+  constructor(config: OpAMPClientHttpConfig) {
+    this.config = config;
+    this.OpAMPInstanceUidString = uuidv7();
+    this.OpAMPInstanceUid = new TextEncoder().encode(
+      this.OpAMPInstanceUidString
+    );
+    this.httpClient = axios.create({
+      baseURL: `http://${this.config.opAMPServerHost}`,
+      headers: {
+        "Content-Type": " application/x-protobuf",
+        Authorization: `DeviceId ${config.instrumentationDeviceId}`,
```

Comment:
I think calling the header `Authorization` is a little misleading.
This gives a sense of secure connection but obviously, it is very easy to manipulate. Maybe something like `X-Odigos-DeviceId`